api/src/main/java/org/openmrs/module/attachments/obs/ImageAttachmentHandler.java

```diff
@@ -49,54 +45,86 @@ protected ComplexData readComplexData(Obs obs, ValueComplex valueComplex, String
 	}
 
 	@Override
-	protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData) {
+	protected boolean deleteComplexData(Obs obs) {
 
-		// We use a temp obs whose complex data points to the file names
-		String fileName = complexData.getTitle();
-		boolean isThumbNailPurged = true;
-		Obs tmpObs = new Obs();
+		// first delete the thumbnail if it exists
+		Boolean isThumbnailPurged = null;
+		String thumbnailDataKey = appendThumbnailSuffix(((ImageHandler) getParent()).parseDataKey(obs));
 
-		if (!isThumbnail(fileName)) {
-			String thumbnailFileName = buildThumbnailFileName(fileName);
-			tmpObs.setValueComplex(thumbnailFileName);
-			isThumbNailPurged = getParent().purgeComplexData(tmpObs);
+		try {
```

Comment:
We also add custom code to access the storage service directly and detail any thumbnail file if it exists when deleting an image obs.